grails-app/services/org/pih/warehouse/inventory/InventorySnapshotService.groovy

```diff
@@ -425,6 +426,50 @@ class InventorySnapshotService {
         log.info "Updated ${results} inventory snapshots for product ${product}"
     }
 
+    /**
+     * Used for product merge feature (when primary product *had not* the same lot as obsolete product)
+     * Change product to primary for rows with given inventory item
+     * */
+    void updateInventorySnapshots(InventoryItem inventoryItem, Product product) {
```

Comment:
We can keep this for now but I'm not overly concerned with inventory snapshot at the moment. Product availability, sure. But it's all of the associations around product that should be given priority at the moment.